app/src/main/java/com/github/sdpteam15/polyevents/model/map/GoogleMapActionHandler.kt

```diff
@@ -73,18 +77,46 @@ object GoogleMapActionHandler {
      * @param locationActivated is location of the visitor activated
      */
     fun onInfoWindowClickHandler(
-        activity: Activity,
+        activity: FragmentActivity,
         lifecycle: LifecycleOwner,
         marker: Marker,
         locationActivated: Boolean
     ) {
-        HelperFunctions.getLoc(activity).observeOnce(lifecycle) {
-            RouteMapHelper.chemin =
-                RouteMapHelper.getShortestPath(it.value!!, marker.tag.toString(), locationActivated)
-                    ?.toMutableList()
-                    ?: mutableListOf()
-            RouteMapHelper.drawRoute()
-        }
+        /**
+         * Display a new ZonePreviewDialog bottom sheet, displaying the zone and the current events
+         * going on. Has 2 buttons, one for showing the itinerary on the map, the other to see all
+         * the events going on in that zone
+         */
+        ZonePreviewBottomSheetDialogFragment.newInstance(
+            zoneId = marker.tag as String,
+            onShowEventsClickListener = {
+                HelperFunctions.changeFragmentWithBundle(
+                    activity,
+                    ZoneEventsFragment::class.java,
+                    bundle = bundleOf(
+                        // marker tag should hold the zone id
+                        ZonePreviewBottomSheetDialogFragment.EXTRA_ZONE_ID to marker.tag,
+                        // marker title should hold the zone name
+                        ZonePreviewBottomSheetDialogFragment.EXTRA_ZONE_NAME to marker.title
+                    ),
+                    addToBackStack = true
+                )
+            },
+            onItineraryClickListener = {
+                // The listener invoked when clicked on the show itinerary button
+                HelperFunctions.getLoc(activity).observeOnce(lifecycle) {
+                    RouteMapHelper.chemin =
+                        RouteMapHelper.getShortestPath(
+                            it.value!!,
+                            marker.tag.toString(),
+                            locationActivated
+                        )
+                            ?.toMutableList()
+                            ?: mutableListOf()
+                    RouteMapHelper.drawRoute()
+                }
+            }
+        ).show(activity.supportFragmentManager, ZonePreviewBottomSheetDialogFragment.TAG)
```

Comment:
I think It would be better if we can in some way put this into the "onPolygonClickListener" of a zone. The user may not be aware that he's supposed to click on the marker that appears on the zone to get its corresponding infos + it requires one more click. You can keep it like that for now but I think we should find a way to show these infos just after the user clicks on the polygon before the last sprint.

Reply:
Agreed! I thought about this too but if we choose to do it this way, the info window displaying the zone name becomes obsolete, should we remove the info window completely?

Reply:
I think we can remove it, it doesn't give any more information about the zone